### ISSUE TITLE
how to config package to work offline?

### ISSUE THREAD

**Issue by OmerDavidson** [jimmywarting/native-file-system-adapter#34]
Because this package is using a CDN for web-streams-polyfill I can't use this package offline.
Is there a way to configure it so it will get that package from npm? So it will fetch it from my private npm registry?

**Comment by jimmywarting** [comment]
it tries to load it from 3 possible places, 
- global context
- cdn 
- or from the optional npm module (but it's mostly meant for if you are using this on the server)

you could try to load your own polyfill and assign it to `globalThis` context
another possible solution could be to pre-catch it

```js
self.addEventListener('install', function(event) {
  event.waitUntil(
    caches.open(cacheName).then(function(cache) {
      return cache.addAll(
        [
          'https://cdn.jsdelivr.net/npm/web-streams-polyfill@3/dist/ponyfill.es2018.mjs'
        ]
      )
    })
  )
})
```

and then do something like:

```js
self.addEventListener('fetch', function(event) {
  event.respondWith(
    caches.match(event.request).then(function(response) {
      return response || fetch(event.request);
    })
  );
});
```


https://developers.google.com/web/ilt/pwa/caching-files-with-service-worker

**Comment by jimmywarting** [comment]
it's also possible to use import maps

```html
<script type="importmap">
{
  "imports": {
    "web-streams-polyfill": "./pkg/main.js"
  }
}
</script>
```

**Comment by milahu** [comment]
fixed by #41
